app/src/main/java/com/epfl/drawyourpath/database/MockDataBase.kt

```diff
@@ -46,73 +53,79 @@ class MockDataBase : Database() {
     }
 
-    override fun setUsername(username: String) {
-        usersDataBase.add(username)
+    override fun setUsername(username: String): CompletableFuture<Boolean> {
+        return isUsernameAvailable(username).thenApply {
+            if (it) {
+                usernameToUserId.put(username, userIdTest)
+                userIdToUsername.put(userIdTest, username)
+            }
+            it
+        }
+    }
+
+    override fun initUserProfile(userModel: UserModel): CompletableFuture<Boolean> {
+        userIdToUserAccount.put(userIdTest, userModel)
+        return CompletableFuture.completedFuture(true)
     }
 
-    override fun setPersonalInfo(
-        username: String,
-        firstname: String,
-        surname: String,
-        dateOfBirth: LocalDate
-    ) {
-        val userPersonalInfo: PersonalInfo = PersonalInfo(firstname, surname, dateOfBirth)
-        usernameToPersonalInfoDataBase.put(username, userPersonalInfo)
+    override fun getUserAccount(userId: String): CompletableFuture<UserModel> {
+        return CompletableFuture.completedFuture(userIdToUserAccount.get(userId))
     }
 
-    override fun setUserGoals(
-        username: String,
-        distanceGoal: Double,
-        timeGoal: Double,
-        nbOfPathsGoal: Int
-    ) {
-        val userGoals: UserGoals = UserGoals(username, distanceGoal, timeGoal, nbOfPathsGoal)
-        usernameToUserGoalsDataBase.put(username, userGoals)
+    override fun getLoggedUserAccount(): CompletableFuture<UserModel> {
+        return getUserAccount(userIdTest)
     }
 
-    override fun setDistanceGoal(userId: String, distanceGoal: Double): CompletableFuture<Boolean> {
-        userIdToDistanceGoal.put(userId, distanceGoal)
-        return CompletableFuture.completedFuture(userIdToDistanceGoal.get(userId) == distanceGoal)
+    override fun setDistanceGoal(distanceGoal: Double): CompletableFuture<Boolean> {
+        if(distanceGoal <= 0.0){
+            throw java.lang.Error("The distance goal can't be less or equal than 0.")
+        }
+        val updatedUser = UserModel(userAuthTest, firstnameTest, surnameTest, dateOfBirthTest, distanceGoal,
+            activityTimeGoalTest, nbOfPathsGoalTest, this)
+        userIdToUserAccount.put(userIdTest, updatedUser)
+        return CompletableFuture.completedFuture(true)
     }
 
-    override fun setActivityTimeGoal(userId: String, activityTimeGoal: Double): CompletableFuture<Boolean> {
-        userIdToActivityTimeGoal.put(userId, activityTimeGoal)
-        return CompletableFuture.completedFuture(userIdToActivityTimeGoal.get(userId) == activityTimeGoal)
+    override fun setActivityTimeGoal(activityTimeGoal: Double): CompletableFuture<Boolean> {
+        if(activityTimeGoal <= 0.0){
+            throw java.lang.Error("The activity time goal can't be less or equal than 0.")
+        }
+        val updatedUser = UserModel(userAuthTest, firstnameTest, surnameTest, dateOfBirthTest, distanceGoalTest,
```

Comment:
Same problem here concerning the `UserModel`, the mock database ends up doing fetches on the firebase backend.

Reply:
resolve: we dont't use the database inside the init user model